geode-core/src/main/java/org/apache/geode/redis/internal/RegionProvider.java

```diff
@@ -187,8 +217,26 @@ public boolean removeKey(ByteArrayWrapper key, RedisDataType type, boolean cance
           return this.stringsRegion.remove(key) != null;
         } else if (type == RedisDataType.REDIS_HLL) {
           return this.hLLRegion.remove(key) != null;
+        } else if (type == RedisDataType.REDIS_LIST) {
+          return this.destroyRegion(key, type);
         } else {
-          return destroyRegion(key, type);
+
+
+          // Check hash
+          ByteArrayWrapper regionName = HashInterpreter.toRegionNameByteArray(key);
+          Region<?, ?> region = this.getRegion(regionName);
+          if (region != null) {
+            region.remove(HashInterpreter.toEntryKey(key));
```

Comment:
We could return true, but that is what the next line after the if statement already returns true
.